test/cli/fixtures/fails/inline-snapshop-inside-loop-update-none.test.ts

```diff
@@ -0,0 +1,9 @@
+import {test, expect} from "vitest";
+
+// this test causes infinite re-run when --watch and --update
+// since snapshot update switches between "foo" and "bar" forever.
```

Comment:
should we do something about this?

Reply:
Yeah, this is not ideal but I couldn't come up with a quick solution from what's available currently.
I think this is rather an edge case compared to OP's case, so we could postpone a proper fix later until someone comes across this. Btw, this behavior is same for current Vitest so that's unchanged. Jest also ends up with infinite re-run for this case.

Mentioning an idea for a proper fix, it would probably require keeping track of all `toMatchInlineSnapshot` call locations, not only failed ones. Currently `parseErrorStacktrace` is not used when `toMatchInlineSnapshot` succeeds, so I need to move up some code from `SnapshotState._addSnapshot` and also adjust states probably.